### PR TITLE
Test restructuring

### PR DESCRIPTION
**Pull request by gusrasch**
# Description

Restructures tests to align them with how code is divided elsewhere
- "unit tests" was a misnomer
- this makes it clearer where stuff should be added in the future

Also removes unneeded test/fixture

Closes: # (issue)

# Checklist:

- [x] I have performed a self-review of my own code
- [x] My code follows the style guidelines of this project
- [x] I have commented my code in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: tests/build_tests/test_build.py [hunk 1]
```diff
@@ -88,30 +88,6 @@ def test_reqs_dependency(self):
...
-
-        :param tmpdir: This is a pytest fixture that creates a temporary directory for us to use
-        :return: The path to the requirements.txt file.
-        """
-        tmpdir.mkdir("tasks").mkdir("subdir").mkdir("subsubdir").join("requirements.txt").write(
-            "magniv"
-        )
-        return str(tmpdir.join("tasks/subdir/subsubdir"))
-
-    def test_subdirreqs_filepath(self, subdir_reqs_file):
```
Comment (jzone3) [inline tests/build_tests/test_build.py:104]: I can't tell, was this moved somewhere or just deleted?

Comment (gusrasch) [reply tests/build_tests/test_build.py:104]: Deleted - if you can tell me what functionality it's covering that isn't already covered by the other tests running save_tasks (the assertion here looks pretty meaningless) I'd add it back

**Review by jzone3 — CHANGES_REQUESTED** [review]
Can you maybe add more descriptive comments in the two folders (maybe even in __init__) to help us decide which test to put in which folder?

Also see comment about removed tests

File: tests/build_tests/test_build.py (hunk 1 above)
Comment (jzone3) [reply tests/build_tests/test_build.py:104]: I think it is supposed to test that nested `requirements.txt` files behave and override as expected but you might be right, not sure this is what happens

**Review by jzone3 — APPROVED** [review]
Love the README

Just make sure we aren't losing a test of `requirements.txt` here. If you are sure then LGTM, ready to merge

File: tests/build_tests/test_build.py (hunk 1 above)
Comment (gusrasch) [reply tests/build_tests/test_build.py:104]: yeah, I see what you're saying, but I agree I don't think that's what's tested. We're passing in `subdir_reqs_file` not the top-level folder (`tasks`) on L112, We can consider re-adding a test like this, but I think this one is so broken it's not worth keeping

**Pull request merged**